yarn-project/ethereum/src/utils.ts

```diff
@@ -64,3 +66,110 @@ function tryExtractEvent<
     }
   }
 }
+
+export function prettyLogViemErrorMsg(err: any) {
+  if (err instanceof BaseError) {
+    const revertError = err.walk(err => err instanceof ContractFunctionRevertedError);
+    if (revertError instanceof ContractFunctionRevertedError) {
+      const errorName = revertError.data?.errorName ?? '';
+      const args =
+        revertError.metaMessages && revertError.metaMessages?.length > 1 ? revertError.metaMessages[1].trimStart() : '';
+      return `${errorName}${args}`;
+    }
+  }
+  return err?.message ?? err;
+}
+
+export function formatViemError(error: any): string {
+  const truncateHex = (hex: string, length = 100) => {
+    if (!hex || typeof hex !== 'string') {
+      return hex;
+    }
+    if (!hex.startsWith('0x')) {
+      return hex;
+    }
+    if (hex.length <= length * 2) {
+      return hex;
+    }
+    return `${hex.slice(0, length)}...${hex.slice(-length)}`;
+  };
```

Comment:
Do we keep the full calldata hex being submitted somewhere in the log? It'd be handy when trying to re-simulate failed tx.